src/components/cart.lineitem.jsx

```diff
@@ -383,6 +499,27 @@ class CartLineItem extends React.Component {
             </span>
           </button>
         </div>
+        {(item._addtocartform) ? (
+          <div className="move-to-cart-btn-col">
+            <button className="ep-btn primary small btn-cart-addToCart" type="button" onClick={this.handleConfiguratorAddToCartBtnClicked}>
+              <span className="btn-text">
+                {intl.get('add-to-cart')}
+              </span>
+            </button>
+          </div>
+        ) : ('')
+        }
+        {(item._dependentoptions && item._dependentoptions[0] && (item._dependentoptions[0]._element || item._dependentlineitems[0]._element)) ? (
+          <div className="configure-btn-col">
+            <button className="ep-btn primary small btn-cart-configureBundle" type="button" onClick={() => this.handleModalOpen()} data-toggle="modal" data-target={`#bundle-configuration-modal-${itemCodeString}`}>
+              <span className="btn-text">
+                {intl.get('configure-bundle')}
+              </span>
+            </button>
+            <AppModalBundleConfigurationMain key={`app-modal-bundle-configuration-main_${itemCodeString}`} handleModalClose={this.handleModalClose} bundleConfigurationItems={item} openModal={openModal} />
```

Comment:
Just want to be sure, REACT is successfully keeping state of which modal is loaded into the DOM at a time and their levels?

Reply:
It is!  When you start going into the nested modals, all open modals remain in the DOM but only one is displayed.  state changes in any child modal are correctly reflected throughout all the parents.